src/connections.c

```diff
@@ -1438,7 +1438,7 @@ static void connection_check_timeout (connection * const con, const unix_time64_
         if (changed)
             con->is_readable = 0;
     }
-    else if (waitevents & FDEVENT_IN) {
+    else if ((waitevents & FDEVENT_IN) || r->state == CON_STATE_READ_POST || r->state == CON_STATE_READ) {
```

Comment:
This is wrong.  If lighttpd is not waiting to read from the socket, there should not be a read timeout.  If lighttpd is streaming the request body to the backend, then lighttpd might have connection in state CON_STATE_READ_POST.